src/network/networkd-link.c

```diff
@@ -3879,8 +3879,9 @@ int link_save(Link *link) {
                 fprintf(f, "REQUIRED_FOR_ONLINE=%s\n",
                         yes_no(link->network->required_for_online));
 
-                fprintf(f, "REQUIRED_OPER_STATE_FOR_ONLINE=%s\n",
-                        strempty(link_operstate_to_string(link->network->required_operstate_for_online)));
+                fprintf(f, "REQUIRED_OPER_STATE_FOR_ONLINE=%s:%s\n",
+                        strempty(link_operstate_to_string(link->network->required_operstate_for_online.min)),
+                        strempty(link_operstate_to_string(link->network->required_operstate_for_online.max)));
```

Comment:
For backward compatibility, please keep the current form when the maximum operstate is not specified. So, maybe, the default value should be `{ LINK_OPERSTATE_DEGRADED, _LINK_OPERSTATE_INVALID }`.

Reply:
I kept backwards compatibility by only printing the max value if it does not equal the default value. Just using `_LINK_OPERSTATE_INVALID` and `strempty` is not sufficient since we'd still be printing the `:` character so we have to do the `if` regardless.